Cannot install this gem on Ubuntu 25.04

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/user/.rvm/gems/ruby-3.1.2/gems/scrypt-3.0.7/ext/scrypt
/home/user/.rvm/rubies/ruby-3.1.2/bin/ruby -I/home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0 -rrubygems /home/user/.rvm/gems/ruby-3.1.2/gems/rake-13.2.1/exe/rake
RUBYARCHDIR\=/home/user/.rvm/gems/ruby-3.1.2/extensions/x86_64-linux/3.1.0/scrypt-3.0.7 RUBYLIBDIR\=/home/user/.rvm/gems/ruby-3.1.2/extensions/x86_64-linux/3.1.0/scrypt-3.0.7
/home/user/.rvm/rubies/ruby-3.1.2/bin/ruby -I/home/user/.rvm/gems/ruby-3.1.2/gems/ffi-compiler-1.3.2/lib/ffi-compiler/fake_ffi -I../../lib/scrypt
/home/user/.rvm/gems/ruby-3.1.2/gems/ffi-compiler-1.3.2/lib/ffi-compiler/exporter.rb ../../lib/scrypt/scrypt_ext.rb ./scrypt_ext.h
mkdir -p x86_64-linux
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/crypto_scrypt.o -c ./crypto_scrypt.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/crypto_scrypt_smix_sse2.o -c
./crypto_scrypt_smix_sse2.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/insecure_memzero.o -c
./insecure_memzero.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/scrypt_ext.o -c ./scrypt_ext.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/sha256.o -c ./sha256.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/scrypt_calibrate.o -c
./scrypt_calibrate.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/scryptenc_cpuperf.o -c
./scryptenc_cpuperf.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/memlimit.o -c ./memlimit.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/crypto_scrypt_smix.o -c
./crypto_scrypt_smix.c
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/warnp.o -c ./warnp.c
./warnp.c: In function 'warnp_setprogname':
./warnp.c:39:16: error: implicit declaration of function 'strdup'; did you mean 'strcmp'? [-Wimplicit-function-declaration]
   39 |         name = strdup(p);
      |                ^~~~~~
      |                strcmp
./warnp.c:39:14: error: assignment to 'char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
   39 |         name = strdup(p);
      |              ^
rake aborted!
Command failed with status (1): [gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std\=c99 -msse -msse2 -D_POSIX_C_SOURCE\=199309L -fPIC -o x86_64-linux/warnp.o
-c ./warnp.c]
/home/user/.rvm/gems/ruby-3.1.2/gems/ffi-compiler-1.3.2/lib/ffi-compiler/compile_task.rb:160:in `block (2 levels) in define_task!'
/home/user/.rvm/gems/ruby-3.1.2/gems/ffi-compiler-1.3.2/lib/ffi-compiler/multi_file_task.rb:24:in `invoke_with_call_chain'
/home/user/.rvm/gems/ruby-3.1.2/gems/rake-13.2.1/exe/rake:27:in `<main>'
Tasks: TOP => default => x86_64-linux/libscrypt_ext.so => x86_64-linux/warnp.o
(See full trace by running task with --trace)

rake failed, exit code 1

Gem files will remain installed in /home/user/.rvm/gems/ruby-3.1.2/gems/scrypt-3.0.7 for inspection.
Results logged to /home/user/.rvm/gems/ruby-3.1.2/extensions/x86_64-linux/3.1.0/scrypt-3.0.7/gem_make.out

  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:95:in `run'
  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/rake_builder.rb:28:in `build'
  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:161:in `build_extension'
  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:195:in `block in build_extensions'
  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `each'
  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in `build_extensions'
  /home/user/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0/rubygems/installer.rb:853:in `build_extensions'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/rubygems_gem_installer.rb:72:in `build_extensions'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/rubygems_gem_installer.rb:28:in `install'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/source/rubygems.rb:201:in `install'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/installer/gem_installer.rb:54:in `install'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/installer/parallel_installer.rb:156:in `do_install'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/installer/parallel_installer.rb:147:in `block in worker_pool'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/worker.rb:62:in `apply_func'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/worker.rb:57:in `block in process_queue'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/worker.rb:54:in `loop'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/worker.rb:54:in `process_queue'
  /home/user/.rvm/gems/ruby-3.1.2/gems/bundler-2.4.15/lib/bundler/worker.rb:90:in `block (2 levels) in create_threads'

An error occurred while installing scrypt (3.0.7), and Bundler cannot continue.

In Gemfile:
  scrypt
```

--- a/ext/scrypt/warnp.c
+++ b/ext/scrypt/warnp.c
@@ -41,7 +41,8 @@ warnp_setprogname(const char * progname)
 		/* No cleanup handler needs to be registered on failure. */
 		return;
 	}
-	strcpy(name, p);
+	strncpy(name, p, strlen(p) + 1);
+	name[strlen(p)] = '\0';  /* Ensure null termination */
 
 	/* If we haven't already done so, register our exit handler. */
 	if (initialized == 0) {

--- a/ext/scrypt/warnp.c
+++ b/ext/scrypt/warnp.c
@@ -36,7 +36,12 @@ warnp_setprogname(const char * progname)
 			p = progname + 1;
 
 	/* Copy the name string. */
-	name = strdup(p);
+	name = malloc(strlen(p) + 1);
+	if (name == NULL) {
+		/* No cleanup handler needs to be registered on failure. */
+		return;
+	}
+	strcpy(name, p);
 
 	/* If we haven't already done so, register our exit handler. */
 	if (initialized == 0) {

